text/0000-temp-components-lib.md

```diff
@@ -0,0 +1,618 @@
+
+1. [Simple components](#simple-components) (Tag, Button, TextLink, etc.)
+2. [Complex compents](#complex-components) (Alert Dialogs, Notifications, Modal, etc.)
+
+Last, there should also be helpers that naturally compliment the experience:
+
+1. [Custom hooks](#custom-hooks)
+2. [Helper functions](#helper-functions)
+
+_:bulb: In addition to React 18, all components should be natively compatible for use
+within React Server Components._
+
+### Simple Components
+
+Simple components should mimick the vanilla JSX design React provides with the Headless-styles
+props extending it (i.e. not manipulating React props or creating "component
+does all" solutions).
+
+For example, a Button should be built something like this:
+
+```typescript
+import { getButtonProps } from '@pluralsight/headless-styles'
+import type { ButtonOptions } from '@pluralsight/headless-styles/types'
+import { type HTMLAttributes, type PropsWithChildren } from 'react'
+
+interface ButtonProps extends ButtonOptions, HTMLAttributes<HTMLButtonElement> {}
+
+function PandoButton(props: PropsWithChildren<ButtonProps>, ref: RefObject) {
+  const {
+    className,
+    disabled,
+    icon,
+    sentiment,
+    size,
+    usage,
+    ...nativeBtnProps
+  } = props
+  const pandoProps = getButtonProps({
+    className,
+    disabled,
+    sentiment,
+    size,
+    usage,
+    icon
+  })
+
+  return (
+    <button {...pandoProps.button} {...nativeProps} ref={ref} />
+  )
+}
+
+export const Button = forwardRef(PandoButton)
+```
+
+By nature, this example uses the native React JSX design while extending the
+Headless-styles props on top of it so it can be used in a way that compliments
+and teaches users about React & Javascript due to not manipluating any props/React APIs.
+
+This is an example of how the Button above can be used:
+
+```typescript
+  <Button type="button" onClick={handleClick} ref={btnRef}>
+    <ShareIcon hidden />
+    Share
+  </Button>
+```
+
+### Complex Components
+
+For more complex components, we should be able to lean into the same design above
+by holding true to our standard of keeping the library designed in a composable way.
+Likewise, most complex components should include a Provider to manage state and/or
+data when needed.
+
+For example, a Tabs family would be composable presentational components that
+include a Provider for state management only, not the data going into them.
+
+This keeps Pando's responsibility clean meaning that Pando should only be
+managing UI states, not the data being used within its components.
+
+_:warning: This would require refactor from the original [Tabs Hooks](https://design.pluralsight.com/docs/reference/react-hooks/useTabs) which are mostly composable, but would need to be
+pushed even further for the ideal design. Likewise, they would need to be moved
+into the new react library for colocation._
+
+An example of the Tabs family in practice, should look something like this:
+
+```jsx
+<Tabs>
```

Comment:
We are curious - is there a way we will be able to respond to tab events via an `onChange` or similar?  [Here is an example from chakra](https://v1.chakra-ui.com/docs/components/disclosure/tabs#tabs-onchange)

Reply:
All components will extend the native React prop and not manipulate the interface, so you will be able to use any prop that React natively allows for all elements. Long-winded "yes" and it will be more flexible by nature than Chakra, MUI, or any other current React component lib that I know of.